lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2505,7 +2511,30 @@ void FIRRTLLowering::initializeRegister(Value reg) {
     addToInitialBlock([&]() {
       emitRandomizePrologIfNeeded();
       circuitState.used_RANDOMIZE_REG_INIT = 1;
-      addToIfDefProceduralBlock("RANDOMIZE_REG_INIT", [&]() { randomInit(); });
+      auto *block = builder.getBlock();
+
+      // Randomized values are assigned to registers in `ifdef
+      // RANDOMIZE_REG_INIT block.
+      auto &op = randomizeRegInitIfOp[block];
+      if (!op)
+        op = builder.create<sv::IfDefProceduralOp>("RANDOMIZE_REG_INIT",
+                                                   [&]() {});
+      runWithInsertionPointAtEndOfBlock(randomInit, op.thenRegion());
+
+      // If the register is async reset, we need to insert extra initialization
+      // in post-randomization so that we can set the reset value to register if
+      // the reset signal is enabled.
+      if (asyncRegResetInitPair.hasValue()) {
+        Value resetSignal, resetValue;
+        std::tie(resetSignal, resetValue) = *asyncRegResetInitPair;
```

Comment:
C++17 should support:

```
Value [resetSignal, resetValue] = *asyncRegResetInitPair;
```

Reply:
Unfortunately, we can not do this until c++20 because `resetValue` is captured into lambda. 
```
auto [resetSignal, resetValue] = *asyncRegResetInitPair;
[&]() { builder.create<sv::BPAssignOp>(reg, resetValue); }
```

https://stackoverflow.com/questions/46114214/lambda-implicit-capture-fails-with-variable-declared-from-structured-binding

Reply:
🥲 Thanks for checking.